Github support added

ikeshav26 opened the pull request:
## Add GitHub deep link support & fix ESLint pre-commit issues

### Summary
Adds GitHub deep-link support to the core library and fixes pre-commit lint failures by using a CommonJS ESLint config compatible with `lint-staged`.

### Changes
- Added `githubHandler` to support GitHub user, repo, issue, and PR URLs.
- Exported and registered the handler in the core index.
- Added `github` to the `Platform` type.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added GitHub deep-link support on iOS and Android for profiles, repositories, pull requests and file blobs.
  * Platform detection now recognizes GitHub so links are detected and routed appropriately.
  * Demo updated with a GitHub example link.

* **Chores**
  * Minor formatting cleanup in project metadata (end-of-file newline).

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ikeshav26 commented:
Thanks! Please let me know if anything else is needed.

mdsaban commented on packages/core/src/types.ts:
```diff
@@ -1,4 +1,4 @@
-export type Platform = 'youtube' | 'linkedin' | 'instagram' | 'unknown';
+export type Platform = 'youtube' | 'linkedin' | 'instagram' | 'spotify' | 'github' | 'unknown';
```
why spotify is here?

mdsaban commented on package.json:
```diff
@@ -27,7 +27,7 @@
   "lint-staged": {
     "**/*.{ts,tsx,js,jsx}": [
-      "eslint --fix",
+      "eslint --fix --config .eslintrc.cjs",
```
let's also revert this change

ikeshav26 commented:
@mdsaban actually first I added Spotify support and going to contribute then I checked there is already one PR for that so I changed it for GitHub I apologise that's my mistake I forgot to remove Spotify from all files . Will fix all this things soon and solve merge conflicts if any.

mdsaban commented on packages/core/src/platforms/github.ts:
```diff
@@ -0,0 +1,51 @@
+
+    // PR's and Blobs
+    if (repo && type && remainder) {
+      if (type === 'pull') {
```
can you use map instead of if/else for this? check other handlers once

mdsaban commented:
Thanks for this @ikeshav26
